ProcessLib/ComponentTransport/ComponentTransportProcess.cpp

```diff
@@ -20,6 +20,61 @@ namespace ProcessLib
 {
 namespace ComponentTransport
 {
+
+void checkMPLProperties(MeshLib::Mesh const& mesh,
+                        ComponentTransportProcessData const& process_data)
+{
+    DBUG("Check the media properties of ComponentTransport process ...");
+
+    std::array<MaterialPropertyLib::PropertyType, 4> const
+        required_properties_medium = {
+            MaterialPropertyLib::PropertyType::porosity,
+            MaterialPropertyLib::PropertyType::transversal_dispersivity,
+            MaterialPropertyLib::PropertyType::longitudinal_dispersivity,
+            MaterialPropertyLib::PropertyType::permeability};
+
+    std::array<MaterialPropertyLib::PropertyType, 2> const
+        required_properties_liquid_phase = {
+            MaterialPropertyLib::PropertyType::density,
+            MaterialPropertyLib::PropertyType::viscosity};
+
+    std::array<MaterialPropertyLib::PropertyType, 3> const
+        required_properties_components = {
+            MaterialPropertyLib::PropertyType::retardation_factor,
+            MaterialPropertyLib::PropertyType::decay_rate,
+            MaterialPropertyLib::PropertyType::molecular_diffusion};
+
+    for (auto const& element : mesh.getElements())
+    {
+        auto const element_id = element->getID();
+
+        auto const& medium = *process_data.media_map->getMedium(element_id);
+        checkRequiredProperties(medium, required_properties_medium);
+
+        // check if liquid phase definition and the corresponding properties
+        // exists
+        auto const& liquid_phase = medium.phase("AqueousLiquid");
+        checkRequiredProperties(liquid_phase, required_properties_liquid_phase);
+
+        // check if components and the corresponding properties exists
+        for (std::size_t component_id = 0;
+             component_id < liquid_phase.numberOfComponents();
```

Comment:
minor: storing the number of components outside of the for-loop would save few instructions...